packages/@guidesmiths/cuckoojs-schematics/src/nestjs-config/README.md

```diff
@@ -0,0 +1,63 @@
+# nestjs-config
+
+## Description
+
+This schematics adds a `Config` module to NestJS projects to validate and load your app's configuration depending on the
+environment your process is executing on.
+
+## How it works
+
+On the app start up, the `config` module will load the configuration file from `src/config/env` depending on the env var 
+(`NODE_ENV` by default) that indicates the environment your process is running on. Configuration defined in `src/config/env/default.ts` 
+will be merged against the one defined in `src/config/env/<ENV>.ts` where `<ENV>` is the current value indicated by the 
+env var defined via the `envVar` option.
+
+## Options
+
+| Option     | Description                                                         | Requiered | Type | Default    |
+|------------|---------------------------------------------------------------------|---|---|------------|
+| `directory` | Root folder of your NestJS project                                  | false | string | `.`        |
+| `envVar  ` | Name of the env var that determines your environment during runtime | false | string | `NODE_ENV` |
+
+## How to use it within a project
+
+Add it to your project running:
+
+```bash
+schematics @onebeyond/cuckoojs-schematics:nestjs-config --directory=. --envVar=NODE_ENV
+```
+
+This will generate a new NestJS module in `src/config` and the necessary `import`s to your `app.module.ts`.
```

Comment:
Perhaps it is worth mentioning that in order for this schematic to work, the nestjs project needs to have the file `./src/app.module.ts`, as it's there where the config will be imported (in new releases we could make this file configurable).